internal/services/notification/notification_test.go

```diff
@@ -0,0 +1,217 @@
+
+func TestRunWebhookDelivery(t *testing.T) {
+	t.Parallel()
+
+	dir := t.TempDir()
+	ctx := context.Background()
+	log := testutil.NewLogger(t)
+
+	ns := setupNotificationService(ctx, t, log, dir)
+
+	t.Logf("starting ns")
+
+	time.Sleep(1 * time.Second)
+
+	// test delivery fail
+
+	runWebhook01 := createRunWebhook(t, ctx, ns)
+	runWebhookDelivery01 := createRunWebhookDelivery(t, ctx, ns, runWebhook01.ID, types.DeliveryStatusNotDelivered)
+
+	if err := ns.handleRunWebhookDelivery(ctx, runWebhookDelivery01.ID); err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	runWebhookDeliveries := getRunWebhookDeliveries(t, ctx, ns)
+	if len(runWebhookDeliveries) != 1 {
+		t.Fatalf("expected %d runWebhook deliveries got: %d", 1, len(runWebhookDeliveries))
+	}
+	if runWebhookDeliveries[0].DeliveryStatus != types.DeliveryStatusDeliveryError {
+		t.Fatalf("expected runWebhook delivery status %q, got %q", types.DeliveryStatusDeliveryError, runWebhookDeliveries[0].DeliveryStatus)
+	}
+
+	// test delivery success
+
+	wr := setupWebhooksReceiver(ctx, t)
+	defer wr.stop()
+
+	t.Logf("starting webhooks client")
+
+	ns.c.WebhookURL = fmt.Sprintf("%s/%s", wr.exposedURL, "webhooks")
+	ns.c.WebhookSecret = "secretkey"
+
+	runWebhook02 := createRunWebhook(t, ctx, ns)
+	runWebhookDelivery02 := createRunWebhookDelivery(t, ctx, ns, runWebhook02.ID, types.DeliveryStatusNotDelivered)
+
+	if err := ns.handleRunWebhookDelivery(ctx, runWebhookDelivery02.ID); err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	runWebhookDeliveries = getRunWebhookDeliveries(t, ctx, ns)
+	if len(runWebhookDeliveries) != 2 {
+		t.Fatalf("expected %d runWebhook deliveries got: %d", 2, len(runWebhookDeliveries))
+	}
+	if runWebhookDeliveries[0].DeliveryStatus != types.DeliveryStatusDeliveryError {
+		t.Fatalf("expected runWebhook delivery status %q, got %q", types.DeliveryStatusDeliveryError, runWebhookDeliveries[0].DeliveryStatus)
+	}
+	if runWebhookDeliveries[1].DeliveryStatus != types.DeliveryStatusDelivered {
+		t.Fatalf("expected runWebhook delivery status %q, got %q", types.DeliveryStatusDelivered, runWebhookDeliveries[1].DeliveryStatus)
+	}
+
+	webhooks, err := wr.webhooks.getWebhooks()
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	if len(webhooks) != 1 {
+		t.Fatalf("expected %d run webhook got: %d", 1, len(webhooks))
+	}
+	if diff := cmp.Diff(webhooks[0].payload, runWebhook02.Payload); diff != "" {
+		t.Fatalf("run webhook payload mismatch (-want +got):\n%s", diff)
+	}
+
+	h256 := hmac.New(sha256.New, []byte(ns.c.WebhookSecret))
+	_, err = h256.Write([]byte(runWebhook02.Payload))
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+	expectedsignature := hex.EncodeToString(h256.Sum(nil))
+
+	if diff := cmp.Diff(webhooks[0].signature, expectedsignature); diff != "" {
+		t.Fatalf("run webhook signature mismatch (-want +got):\n%s", diff)
+	}
+
+	// test runWebhookDelivery01 just delivered
+	err = ns.handleRunWebhookDelivery(ctx, runWebhookDelivery01.ID)
```

Comment:
You should not call handleRunWebhookDelivery but instead call runWebhookDeliveriesHandler so you'll check that the query and all the logic works.